From b456cad6d7335065d6667d153489c1c7da5e3ffc Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Mon, 6 Aug 2012 23:23:12 +0300
Subject: [PATCH 276/311] conf/obj.c: use C2_IN() in c2_conf__readdir_post()

c2_conf__readdir_post() never receives negative values of `retval'.

+ obj_ops.c: Add missing `)' to dir_readdir().
---
 conf/obj.c     | 4 +++-
 conf/obj_ops.c | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/conf/obj.c b/conf/obj.c
index 05f0756..4f530ec 100644
--- a/conf/obj.c
+++ b/conf/obj.c
@@ -24,6 +24,7 @@
 #include "conf/obj.h"
 #include "lib/arith.h"  /* C2_CNT_INC, C2_CNT_DEC */
 #include "lib/cdefs.h"  /* ergo */
+#include "lib/misc.h"   /* C2_IN */
 
 /**
  * @page conf DLD of configuration caching
@@ -467,7 +468,8 @@ bool c2_conf__readdir_pre(const struct c2_conf_obj *dir,
 bool c2_conf__readdir_post(int retval, const struct c2_conf_obj *dir,
 			   const struct c2_conf_obj *entry)
 {
-	return retval <= C2_CONF_DIRMISS &&
+	return C2_IN(retval,
+		     (C2_CONF_DIREND, C2_CONF_DIRNEXT, C2_CONF_DIRMISS)) &&
 		(retval == C2_CONF_DIRNEXT ?
 		 (entry != NULL && belongs(entry, dir) && entry->co_nrefs > 0) :
 		 entry == NULL);
diff --git a/conf/obj_ops.c b/conf/obj_ops.c
index 93205ec..2a53a8d 100644
--- a/conf/obj_ops.c
+++ b/conf/obj_ops.c
@@ -169,7 +169,7 @@ static int dir_readdir(struct c2_conf_obj *dir, struct c2_conf_obj **pptr)
 	 *     ret = C2_CONF_DIRNEXT;
 	 * }
 	 *
-	 * C2_POST(c2_conf__readdir_post(ret, dir, *pptr);
+	 * C2_POST(c2_conf__readdir_post(ret, dir, *pptr));
 	 * return ret;
 	 */
 	XXX;
-- 
1.8.3.2

